cmd/earthly/main.go

```diff
@@ -1123,20 +1116,23 @@ func (app *earthlyApp) getAddressesForTCP(context *cli.Context) (addresses, erro
 		return addresses{}, err
 	}
 
-	lrURL, err := parseAndvalidateURL(app.cfg.Global.LocalRegistryHost)
-	if err != nil {
-		return addresses{}, err
+	if buildkitd.IsLocal(app.buildkitHost) && app.cfg.Global.LocalRegistryHost != "" {
+		// Local registry only matters when local, and specified.
+		lrURL, err := parseAndvalidateURL(app.cfg.Global.LocalRegistryHost)
+		if err != nil {
+			return addresses{}, err
+		}
+		if bkURL.Hostname() != lrURL.Hostname() {
+			app.console.Warnf("Buildkit and Local Registry URLs are pointed at different hosts (%s vs. %s)", bkURL.Hostname(), lrURL.Hostname())
+		}
```

Comment:
No checks and no validation if local. We assume that registry code downstream is smart enough.

An alternative here is if its non-local we return an empty string for the address.

Reply:
Wondering if we should write out an error (or at least a warning?) if it's a remote buildkit and the local registry address is set.

Reply:
The more I think about it, the more I would like to return an empty string to force the issue when non local.

> Wondering if we should write out an error (or at least a warning?) if it's a remote buildkit and the local registry address is set.

I could see people having it set for local dev; and then pointing at a big buildkitd in the sky. It would be annoying for a config file value to prevent this easy switching. A warning seems best to me? WYT @vladaionescu ?

Reply:
Yeah, I can see how that can be annoying.

I wonder if the config should be a bit different to make the switch easier, without the gotcha implications. Maybe in the future, we can have a single host + port and we multiplex the different functionalities through that (buildkit, debugger, registry).

In the meantime, I think either a warning or no message is fine, given that it's anyway called **local** registry.

Reply:
Warning added.